### PR TITLE
task 004 completed 

### PR DESCRIPTION
**Pull request by elanghel**
apply Clear Button style : backgroundColor, backgroundColor when highlighted, text color and border radius

### CODE REVIEW

**Review by Vasile95 — APPROVED** [review]

File: GithubPractice/ViewController.swift [hunk 1]
```diff
@@ -26,6 +26,7 @@ class ViewController: UIViewController, UITextFieldDelegate {
         jobPositionField.delegate = self
         
         saveButton.layer.cornerRadius = 8
+        clearButton.layer.cornerRadius = 8
```
Comment (vtoloconnicov) [inline GithubPractice/ViewController.swift]: if all the highlight buttons have same corner radius in the app, better set it in the HighlightButton class

Comment (elanghel) [reply GithubPractice/ViewController.swift]: okay

**Comment by mpetrenco** [comment]
Can we:
- Extract the `UITextFieldDelegate` in a separate extension;
- Make sure that we don't have extra spacing between methods;
- Can we just use backgroundColor instead of a custom one:
```
@IBInspectable var normalBackgroundColor: UIColor? {
        didSet {
            backgroundColor = normalBackgroundColor
        }
    }
 ```

**Review by mpetrenco — CHANGES_REQUESTED** [review]
Just a couple of comments from me 👍

File: GithubPractice/ViewController.swift (hunk 1 above)
Comment (Vasile95) [inline GithubPractice/ViewController.swift]: please move all numbers to constants

**Review by Vasile95 — CHANGES_REQUESTED** [review]

File: GithubPractice/ViewController.xib [hunk 2]
```diff
@@ -88,12 +88,23 @@
...
                             <constraints>
                                 <constraint firstAttribute="height" constant="48" id="lk4-kb-eqp"/>
                             </constraints>
-                            <state key="normal" title="Clear"/>
+                            <state key="normal" title="Clear">
+                                <color key="titleColor" red="1" green="1" blue="1" alpha="1" colorSpace="calibratedRGB"/>
+                            </state>
+                            <userDefinedRuntimeAttributes>
+                                <userDefinedRuntimeAttribute type="color" keyPath="normalBackgroundColor">
+                                    <color key="value" red="0.5450980392" green="0.57647058819999997" blue="0.63921568630000003" alpha="1" colorSpace="calibratedRGB"/>
```
Comment (Vasile95) [inline GithubPractice/ViewController.xib]: https://medium.com/@bhavneetsingh08/apples-new-xcode-feature-named-color-371f59c37137
here is an example on how to add custom colors to the project.

**Comment by Vasile95** [comment]
please fix conflicts

**Comment by elanghel** [comment]
> * Can we just use backgroundColor instead of a custom one:
> 
> ```
> @IBInspectable var normalBackgroundColor: UIColor? {
>         didSet {
>             backgroundColor = normalBackgroundColor
>         }
>     }
> ```


This is because I set UIButton as HighlightButton and this class has the following attributes: normalBackgroundColor and highlightedBackgroundColor.

Screen Shot 2021-03-18 at 11 03 47 AM

 I have to set the background color as normalBackgroundColor otherwise at the launching the the button is white. 

**So I would set a default color to the button using attributes and this code could be deleted :**
` {
        didSet {
            backgroundColor = normalBackgroundColor
        }
    }` 


The result: 
Screen Shot 2021-03-18 at 11 08 18 AM

saveButton doesn't have an default color, but clearButton has one.

File: GithubPractice/ViewController.xib (hunk 2 above)
Comment (elanghel) [reply GithubPractice/ViewController.xib]: Done, in the next comit it will be seen.

File: GithubPractice/ViewController.swift (hunk 1 above)
Comment (elanghel) [reply GithubPractice/ViewController.swift]: Done

**Comment by elanghel** [comment]
> Can we:
> 
> * Extract the `UITextFieldDelegate` in a separate extension;
> * Make sure that we don't have extra spacing between methods;
> * Can we just use backgroundColor instead of a custom one:
> 
> ```
> @IBInspectable var normalBackgroundColor: UIColor? {
>         didSet {
>             backgroundColor = normalBackgroundColor
>         }
>     }
> ```

**Review by vtoloconnicov — APPROVED** [review]

File: GithubPractice/Colors.xcassets/Dark Green.colorset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Vasile95) [inline GithubPractice/Colors.xcassets/Dark Green.colorset/Contents.json:1]: can you remove spaces from color names? please

File: GithubPractice/ViewController.swift [hunk 4]
```diff
@@ -16,45 +16,36 @@ class ViewController: UIViewController, UITextFieldDelegate {
...
         lastNameField.delegate = self
         jobPositionField.delegate = self
-        
-        saveButton.layer.cornerRadius = 8
-        
     }
     
     @IBAction func didClickSave(_ sender: Any) {
-        defaults.set(firstNameField.text, forKey: "firstName")
-        defaults.set(lastNameField.text, forKey: "lastName")
```
Comment (Vasile95) [inline GithubPractice/ViewController.swift]: hehe, don't remove my code

**Review by Vasile95 — APPROVED** [review]

**Review by mpetrenco — APPROVED** [review]

**Pull request merged**